dbus/safelauncher.go

```diff
@@ -0,0 +1,81 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package dbus
+
+import (
+	"fmt"
+	"net/url"
+	"os/exec"
+
+	"github.com/godbus/dbus"
+	"github.com/snapcore/snapd/strutil"
+)
+
+const safeLauncherIntrospectionXML = `
+<interface name='io.snapcraft.SafeLauncher'>
```

Comment:
Can we please rename this type to Launcher and the file to launcher.go.  Originally it was important to have Safe in here because it was a general namespace. The safety implications are now implied with this being under snapcraft.io, in that safety is a concern across all such external interactions, so we don't need to call it out explicitly.

We may also end up putting this behind an interface some day, so the name would also be less right in that case.